polaris-react/src/components/IndexTable/IndexTable.scss

```
@@ -232,6 +276,10 @@ $loading-panel-height: 53px;
   font-size: var(--p-font-size-75);
   white-space: nowrap;
   border: 0;
+
+  #{$se23} & {
+    padding: var(--p-space-1_5-experimental);
```

Looks like line-height for TableHeading is 20px in Storybook but 16px on the Figma. Can probably supplement this with:

```suggestion
    padding: var(--p-space-1_5-experimental);
    line-height: var(--p-font-line-height-1);
```